internal/utils.go

```diff
@@ -0,0 +1,36 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package internal
+
+type LockMap interface {
+	Iter(func(key, val string))
+	Len() int
+	Clear()
+}
+
+// ReadOnlyLockMap is a read only version of LockMap which avoids using mutexes as the caller guarantees no writes.
+type ReadOnlyLockMap struct {
+	m map[string]string
+}
+
+func NewReadOnlyLockMap(m map[string]string) *ReadOnlyLockMap {
+	return &ReadOnlyLockMap{m: m}
+}
+
+func (r *ReadOnlyLockMap) Iter(f func(key string, val string)) {
+	for k, v := range r.m {
+		f(k, v)
+	}
+}
+
+func (r *ReadOnlyLockMap) Len() int {
+	return len(r.m)
+}
+
+// Clear does panics as it is not safe to clear a ReadOnly map
+func (r *ReadOnlyLockMap) Clear() {
+	panic("Not safe to call Clear on a ReadOnlyLockMap")
+}
```

Comment:
:thinking: I'm usually not a fan of types that *pretend* to implement an interface.

Do we need an interface at all? Is there harm in just having a single `LockMap` struct type that implements all of these? The performance hit from an uncontested rlock should be trivial.

Alternatively, if we think it's really important, I'd rather have a `ROMap` interface that only specifies `Iter` and `Len`. Then the contribs can use *that* type, and both the `ReadOnlyLockMap` with `Clear()` removed (and probably renamed to `ReadOnlyMap`) and the `globalconfig.headerTagMap` will satisfy that interface.